test/InventoryItem.js

```diff
@@ -0,0 +1,277 @@
+/* global describe it before after */
+// During the test the env variable is set to test
+process.env.NODE_ENV = 'test';
+const InventoryItem = require('../api/models/InventoryItem');
+// Require the dev-dependencies
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const constants = require('../api/constants');
+const { OK, BAD_REQUEST, UNAUTHORIZED, NOT_FOUND } = constants.STATUS_CODES;
+
+let app = null;
+
+const expect = chai.expect;
+// tools for testing
+const tools = require('../util/testing-utils/tools.js');
+const {
+  setTokenStatus,
+  resetMock,
+  restoreMock
+} = require('./mocks/TokenValidFunctions');
+
+chai.should();
+chai.use(chaiHttp);
+
+describe('InventoryItem', () => {
+  before(done => {
+    app = tools.initializeServer();
+    // Before each test we empty the database
+    tools.emptySchema(InventoryItem);
+    done();
+  });
+
+  // What does these do
+  after(done => {
+    restoreMock();
+    tools.terminateServer(done);
+  });
+
+  beforeEach(() => {
+    setTokenStatus(false);
+  });
+
+  afterEach(() => {
+    resetMock();
+  });
+
+  const token = '';
+  let itemName = '';
+  const VALID_NEW_ITEM = {
+    name: 'Chips',
+    price: 0.75,
+    stock: 25,
+    category: 'Food',
+    description: 'Food that you can eat',
+    picture: '%&(&%##$^&(&#@'
+  };
+  const ITEM_WITH_INVALID_TOKEN = {
+    token: 'invalid'
+  };
+  const ITEM_WITHOUT_REQUIRED_FIELDS = {
+    name: 'chocolate'
+  };
+  const ITEM_WITH_INVALID_NAME = {
+    name: 'bob'
+  };
+  const UPDATED_ITEM = {
+    name: 'Chips',
+    price: 0.5,
+    stock: 15,
+    category: 'Snack',
+    description: 'A snack that you can eat',
+    picture: ')(^#!@$%*(*'
+  };
+
+  describe('/POST addItem', () => {
+    it('Should return 403 when an invalid token is supplied', done => {
+      chai
+        .request(app)
+        // can you explain this routing
```

Comment:
@ThaiQ

Reply:
Sorry have to take out those comments...Already asked thai those questions